File: Java/MIMIC/src/ejemplosencillo/EDA/Item.java
package ejemplosencillo.EDA;

import ejemplosencillo.base.IFuncionObjetivo;
import java.util.LinkedList;

public class Item implements Comparable<Item>, Cloneable{
    private LinkedList<Byte> values = null;
    private Integer fitness;

    public LinkedList<Byte> getValues() {
        return values;
    }
    public Integer getFitness() {
        return fitness;
    }
    
    public Item(){}
    private Item(Integer fitness, LinkedList<Byte> values){
        this.fitness=fitness;
        this.values = (LinkedList<Byte>) values.clone();
    }
    
    public void Generar(LinkedList<Float> probabilidades){
        this.values = new LinkedList<>();
        for (Float prob: probabilidades){
            if ((float)Math.random()<prob){
                this.values.add((byte) 1);
            } else {
                this.values.add((byte) 0);
            }
        }
    }
    
    public void Evaluar(IFuncionObjetivo funcion){
        this.fitness = funcion.evaluar(values);
    }

    @Override
    public int compareTo(Item o) {
        return this.fitness.compareTo(o.fitness);
    }
    
    @Override
    public Item clone(){
        return new Item(this.fitness, this.values);
    }

    @Override
    public String toString() {
        return fitness + "=[" + values + "]";
    }
    
    
}

File: kotlin/MIMIC/src/ejemplosencillo/EDA/Item.kt
package ejemplosencillo.EDA

import ejemplosencillo.base.IFuncionObjetivo
import java.util.LinkedList

class Item : Comparable<Item>, Cloneable {
    var values: LinkedList<Byte>? = null
        private set
    var fitness: Integer? = null
        private set

    constructor() {}
    private constructor(fitness: Integer, values: LinkedList<Byte>) {
        this.fitness = fitness
        this.values = values.clone() as LinkedList<Byte>
    }

    fun Generar(probabilidades: LinkedList<Float>) {
        this.values = LinkedList()
        for (prob in probabilidades) {
            if (Math.random() as Float<prob) {
                this.values!!.add(1.toByte())
            } else {
                this.values!!.add(0.toByte())
            }
        }
    }

    fun Evaluar(funcion: IFuncionObjetivo) {
        this.fitness = funcion.evaluar(values)
    }

    @Override
    fun compareTo(o: Item): Int {
        return this.fitness!!.compareTo(o.fitness)
    }

    @Override
    fun clone(): Item {
        return Item(this.fitness, this.values!!)
    }

    @Override
    fun toString(): String {
        return fitness + "=[" + values + "]"
    }


}
